Model/ProductExporter.php

```diff
@@ -0,0 +1,402 @@
+                $product->getShortDescription(),
+                $product->getDescription(),
+                $product->getUrlKey(), // slug
+                $product->getMetaTitle(),
+                $product->getMetaKeyword(),
+                $product->getMetaDescription(),
+                $productData['is_active'],
+                $stockData['is_in_stock'],
+                $stockData['stock_qty'],
+                $stockData['is_manage_stock'],
+                $stockData['is_backorder_enabled'],
+                $taxData['vat_rate'] ?? null,
+                $taxData['vat_iso2'] ?? null,
+                $taxData['vat_rate_alias'] ?? null,
+                $this->storeManager->getStore()->getCurrentCurrencyCode(),
+                $this->productHelper->getTaxPrice($product, $productPrice, false), // price excl. tax
+                $this->productHelper->getTaxPrice($product, $productPrice, true), // price incl. tax
+                $this->productHelper->getTaxPrice($product, $productSpecialPrice, false), // product.product_discount_price.ppu - Discount price
+                $this->productHelper->getTaxPrice($product, $productSpecialPrice, true), // product.product_discount_price.ppu_wt - Discount price with VAT
+                null, // product.product_purchase_price.ppu - Purchase price
+                null, // product.product_purchase_price.ppu_wt - Purchase price with VAT
+                null, // product.product_bottom_price.ppu - Bottom price
+                null, // product.product_bottom_price.ppu_wt - Bottom price with VAT
+                $this->productHelper->getTaxPrice($product, $productCostPrice, false), // product.product_cost_price.ppu - Cost price
+                $this->productHelper->getTaxPrice($product, $productCostPrice, true), // product.product_cost_price.ppu_wt - Cost price with VAT
+                null, // product.configurable_product_kind.alias - Product kind alias
+                null, // product.configurable_product.sku - Configurable product sku
+                $categoryData['category_name'] ?? null,
+                $categoryData['category_url'] ?? null,
+                null, // extra_category_slugs - Extra Category slugs
+                null, // extra_label_slugs - Extra Label slugs
+                $productData['attribute_set_name'],
+                null, // attribute_set_alias - Attribute set alias
+                $productData['is_salable'],
+                null, // shop_products.main.relation_limited - Sales relation limited
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                $productData['image_url']
+            ];
```

Comment:
```suggestion
            ];
            
            yield $row;
```

use [generator](https://www.php.net/manual/en/language.generators.syntax.php) this way you can store only on row in the memory